subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -248,6 +268,64 @@ public void renderNode(StyledTextOutput target, RenderableDependency node, boole
         output.println(" option.");
     }
 
+    private void handleResolutionError(Throwable cause, StyledTextOutput output) {
+        if (cause instanceof VersionConflictException) {
+            handleConflict((VersionConflictException) cause, output);
+        } else if (cause instanceof LockOutOfDateException) {
+            handleOutOfDateLocks((LockOutOfDateException) cause, output);
+        } else {
+            // Fallback to failing the task in case we don't know anything special
+            // about the error
+            throw UncheckedException.throwAsUncheckedException(cause);
+        }
+    }
+
+    private void handleOutOfDateLocks(LockOutOfDateException cause, StyledTextOutput output) {
+        List<String> errors = cause.getErrors();
+        output.text("The dependency locks are out-of-date:");
+        output.println();
+        for (String error : errors) {
+            output.text("   - " + error);
+            output.println();
+        }
+        output.println();
+    }
+
+    private void handleConflict(VersionConflictException conflict, StyledTextOutput output) {
+        for (List<ModuleVersionIdentifier> moduleVersionIdentifiers : conflict.getConflicts()) {
+            boolean matchesSpec = hasVersionConflictOnRequestedDependency(moduleVersionIdentifiers);
```

Comment:
Would there be any downside to reporting the exception even in the case where it doesn't match the requested dependency? 

With this check it seems like we'll be silently ignoring other conflicts. I guess this is probably OK since the user cares about a particular dependency. But i'm not sure the added code is worth it.

Reply:
I added it because in real dependency graphs, you can have a lot of conflicts and it kind of pollutes the output when you focus on a specific one.